agot-bg-game-server/src/client/game-state-panel/PlayerMusteringComponent.tsx

```diff
@@ -148,12 +170,15 @@ export default class PlayerMusteringComponent extends Component<GameStateCompone
 
         const musterings: Mustering[] = this.musterings.tryGet(this.selectedRegion, []);
 
-        musterings.push({
-            from: rule.from,
-            to: rule.to,
-            region: rule.region
-        });
+        if (rule.from) {
+            rule.affectedUnit = this.props.gameState.getUnitsLeftToUpgrade(this.musterings, rule.region)[0];
+            rule.affectedUnit.upgradedType = rule.to;
+        } else {
+            rule.affectedUnit = this.props.gameState.game.createUnit(rule.region, rule.to, this.props.gameState.house);
```

Comment:
We never have called `createUnit` client-side yet but right now I see no other option when we want to show the mustered units.